gradio/external.py

```diff
@@ -417,6 +419,35 @@ def get_spaces(model_name, api_key, alias, **kwargs):
         return get_spaces_blocks(model_name, config)
 
 
+async def get_pred_from_ws(
+    websocket: websockets.WebSocketClientProtocol, data: str
+) -> Dict[str, Any]:
+    completed = False
+    while not completed:
+        msg = await websocket.recv()
+        resp = json.loads(msg)
+        if resp["msg"] == "queue_full":
+            raise exceptions.Error("Queue is full! Please try again.")
+        elif resp["msg"] == "send_data":
+            await websocket.send(data)
+        completed = resp["msg"] == "process_completed"
+    return resp["output"]
+
+
+def get_ws_fn(ws_url):
+    async def ws_fn(data):
+        async with websockets.connect(ws_url, open_timeout=10) as websocket:
+            return await get_pred_from_ws(websocket, data)
+
+    return ws_fn
+
+
+def use_websocket(config):
+    return config["enable_queue"] and version.parse(
```

Comment:
I believe older versions of Gradio did not include the `version` in the config, and so it would be good to check for that first so that a `KeyError` is not thrown

Reply:
Good point!